### ISSUE TITLE
Can't connect to Azure

### ISSUE THREAD

**Issue by bbonn** [xebialabs-community/xld-azure-app-services-plugin#2]
Hey guys,

I am having trouble making any connection to azure using the new plugin.   I realize that may no be your problem :) but was looking for a little help.   I thought perhaps it was a firewall problem.  I followed the directions here (https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-create-service-principal-portal) and using those creds (except for the ftp user) I am able to connect.  First does the check connection use the ftp user, and if so does it have to be in any special format?    Any tips, I can't seem to make headway.  

Error I get when Checking connection:
Error while executing script [azure_app_services/check_connection.py]: java.util.concurrent.ExecutionException: java.util.concurrent.ExecutionException: java.net.ConnectException: Connection refused: connect in <script> at line number 21
com.xebialabs.platform.script.jython.JythonException: Error while executing script [azure_app_services/check_connection.py]: java.util.concurrent.ExecutionException: java.util.concurrent.ExecutionException: java.net.ConnectException: Connection refused: connect in <script> at line number 21 

Thanks!
Bernie

**Comment by jdewinne** [comment]
Hey Bernie,

I might be mistaken, but I think the `check_connection` uses the `clientId` and `clientKey`: https://github.com/xebialabs-community/xld-azure-app-services-plugin/blob/master/src/main/jython/azure_app_services/client/__init__.py#L87

Regards,
Joris

**Comment by bbonn** [comment]
Hey Joris thanks for the reply.   

The client ID and clientkey are 2 values I have validated in power shell, on the same machine.   I followed this sample [here](https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-authenticate-service-principal#provide-credentials-through-powershell) in PS and it worked using the same clientid and client key.   Anything else I can look for?

Thanks again,
Bernie

**Comment by bbonn** [comment]
Hey Guys,

Do you have any tips for troubleshooting this?   Using the same clientId and clientKey and tenentId in the below power shell command connects, but trying the Check Connection in XLD fails with the java connection refused?  Not sure how to figure out what is going on?   Thanks.  

$creds = Get-Credential
Login-AzureRmAccount -Credential $creds -ServicePrincipal -TenantId {tenant-id}

**Comment by jdewinne** [comment]
Did you also follow the instructions as outlined in the [readme](https://github.com/xebialabs-community/xld-azure-app-services-plugin#azure-connection-information)?

I haven't validated yet, but that should work normally.

**Comment by bbonn** [comment]
Hey Joris,

Yes I did, that was how I got the values I used to login via power shell.

**Comment by jdewinne** [comment]
I've just tried this without any problem.
You've also added the app to the correct role with subscription? And also configured the [service principal](https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-create-service-principal-portal)?

**Comment by bbonn** [comment]
Thanks Joris, 
Yes that is what I followed to get the credentials I used to login via power shell.   I can go through it again I suppose, but i think I followed that page you referenced to the letter to get the creds.

**Comment by jdewinne** [comment]
Or maybe a copy/paste error on one of the values?

**Comment by bbonn** [comment]
maybe, but I copied and pasted out of XLD into power shell command and it worked.

**Comment by bbonn** [comment]
Hey Joris,

Any way you could share your creds that you know work (clientID, clientKey, and tenantId)?  Is it just a sample app?   That way I would know for sure it is not something else.

**Comment by jdewinne** [comment]
I can't share, but we could setup a screensharing session to test this. Would that work?

**Comment by bbonn** [comment]
Yeah that would be great.  I think I will try from home to make sure it is not our firewall or something.   If that doesn't clear it up I'll set something up for a screenshare.   

Thanks Joris!

**Comment by bbonn** [comment]
Arrrgh.  it was the corporate firewall.   Sorry about all that, should have tried at home early on.  I was tricked by the fact that I was able to connect via power shell.  

So that being said is it possible to add a some new properties for proxy creds?    I realize this is a community plugin and we can do some of it, but you guys seem so much closer to it and also wondering if there are plans to get it into XLD proper.  

Thanks again Joris sorry to waste your time with this goose chase :)
Bernie

**Comment by bbonn** [comment]
Hey joris,

Any idea what port this is using, maybe it is as simple as asking our security folks to open it up.

**Comment by jdewinne** [comment]
I would expect it to use port 80/443.
Also, part of this plugin is also covered in XLD 7.x. However, that at the same time causes some type clashes with this plugin. So depending on what functionality you need, we should check how to upgrade I think.

**Comment by bbonn** [comment]
Thanks Joris,

I was doing a bit more research on this and it seems like the 1.0 version of adal4j-1.0.0.jar does not have any option for proxy.  If the plugin used 1.1.3 i 'think' it could take some System.setProperty proxy type arguments.   

I think the functionality we need at this point is deploying web-apps, but it is very early and I was just trying to get something simple to work to prove out the plugin, but the proxy issue may be a tough one to get by.   Do you think that is something you guys would tackle?

We may be looking to upgrade XLD to 7 in the next month or 2.

Thanks,
Bernie

**Comment by jdewinne** [comment]
Could be. The [official azure plugin](https://docs.xebialabs.com/xl-deploy-xld-azure-plugin/7.0.x/azurePluginManual.html) is based on the python libs. However, it doesn't contain yet the app services. So maybe we should get that on the backlog? Can you submit an official Feature request?

**Comment by bbonn** [comment]
Hey Joris,
Just so I am clear.  I should make a feature request to add proxy functionality for the Official Azure plugin.  Are you guys planning on moving the app services deployment stuff from this plugin in there soon?   That plugin looks like it only handles infrastructure type actions at this point.   
Thanks!

**Comment by jdewinne** [comment]
I think first step should be to make them compatible. At this moment they clash.
Next step might be to reuse the same types.

**Comment by bbonn** [comment]
OK thanks Joris.   I feature requested against the official one, but it would be cool if we could get this one updated so we could use it here at Paychex in the near future (i.e. behind company firewall).  I would take a whack at it but I am not much of a developer.  I could possibly get one of the guys on the team to take a look and maybe contribute back.
